context.R

```diff
@@ -67,6 +67,84 @@ context <- function(testcases={}, preExec={}) {
     )
 }
 
+
+contextWithParsing <- function(testcases={}, preExec={}) {
+    testcases <- substitute(testcases)
+
+    get_reporter()$start_context()
+    do_exit <- TRUE
+    on.exit({
+        if(do_exit) {
+            get_reporter()$end_context()
+        }
+    })
+
+    # remove unnamed list items and throw a warning
+    filtered_testcases <- testcases[!duplicated(names(testcases)) & names(testcases)!=""]
+    if(length(filtered_testcases) > length(testcases)){
+        get_reporter()$add_message("warning: There are duplicate names and/or unamed testcases found. Note that these won't be used for evaluation.")
```

Comment:
```suggestion
        get_reporter()$add_message("warning: There are duplicate names and/or unnamed testcases found. Note that these won't be used for evaluation.")
```

This also feels like something that should result in an internal error, since the teacher made a mistake when writing testcases. This message should also only be available to staff and up.